x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/relevance_tuning_logic.ts

```diff
@@ -155,4 +287,270 @@ export const RelevanceTuningLogic = kea<
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+
+      const url = `/api/app_search/engines/${engineName}/search_settings/details`;
+
+      try {
+        const response = await http.get(url);
+        actions.onInitializeRelevanceTuning(response);
+      } catch (e) {
+        flashAPIErrors(e);
+      }
+    },
+    getSearchResults: async (_, breakpoint) => {
+      await breakpoint(250);
+
+      const { engineName } = EngineLogic.values;
+      const { http } = HttpLogic.values;
+      const query = values.query;
+      const { search_fields: searchFields, boosts } = removeBoostStateProps(values.searchSettings);
+      if (!query) {
+        return actions.clearSearchResults();
+      }
+      actions.setResultsLoading(true);
+      const url = `/api/app_search/engines/${engineName}/search_settings_search`;
+
+      try {
+        const response = await http.post(url, {
+          query: {
+            query,
+          },
+          body: JSON.stringify({
+            boosts: isEmpty(boosts) ? undefined : boosts,
+            search_fields: isEmpty(searchFields) ? undefined : searchFields,
+          }),
+        });
+
+        actions.setSearchResults(response.results);
+      } catch (e) {
+        flashAPIErrors(e);
+      }
+    },
+    onSearchSettingsSuccess: ({ searchSettings }) => {
+      actions.setSearchSettingsResponse(searchSettings);
+      actions.getSearchResults();
+      window.scrollTo(0, 0);
+    },
+    onSearchSettingsError: () => {
+      window.scrollTo(0, 0);
+    },
+    updateSearchSettings: async () => {
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+
+      const url = `/api/app_search/engines/${engineName}/search_settings`;
+
+      try {
+        const response = await http.put(url, {
+          body: JSON.stringify(removeBoostStateProps(values.searchSettings)),
+        });
+        setSuccessMessage(UPDATE_SUCCESS_MESSAGE);
+        actions.onSearchSettingsSuccess(response);
+      } catch (e) {
+        flashAPIErrors(e);
+        actions.onSearchSettingsError();
+      }
+    },
+    resetSearchSettings: async () => {
+      if (window.confirm(RESET_CONFIRMATION_MESSAGE)) {
+        const { http } = HttpLogic.values;
+        const { engineName } = EngineLogic.values;
+
+        const url = `/api/app_search/engines/${engineName}/search_settings/reset`;
+
+        try {
+          const response = await http.post(url);
+          setSuccessMessage(DELETE_SUCCESS_MESSAGE);
+          actions.onSearchSettingsSuccess(response);
+        } catch (e) {
+          flashAPIErrors(e);
+          actions.onSearchSettingsError();
+        }
+      }
+    },
+    toggleSearchField: ({ name, disableField }) => {
+      const { searchSettings } = values;
+      const { search_fields: searchFields } = searchSettings;
+
+      actions.setSearchSettings({
+        ...searchSettings,
+        boosts: searchSettings.boosts,
```

Comment:
Out of curiosity, why do we need to re-specify `boosts` if we're already cloning `searchSettings`? Am I wrong, or can we just remove this line?

```suggestion
```

Reply:
We don't, good catch

Reply:
f16fd97519c0eb8f6e921fe30da70aea10fb26be